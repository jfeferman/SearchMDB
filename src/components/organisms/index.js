export { default as PersonCard } from './PersonCard';
export { default as SearchBlock } from './SearchBlock';
export { default as MovieCard } from './MovieCard';
export { default as TvCard } from './TvCard';
