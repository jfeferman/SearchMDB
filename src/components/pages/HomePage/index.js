export { default } from './HomePage';
export { QueryContext } from './context';
