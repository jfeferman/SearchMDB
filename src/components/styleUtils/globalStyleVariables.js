export const colors = {
  white: '#ffffff',
  black: '#000000',
  blue: '#176DDC',
  hoverBlue: '#0B356C',
  lightGray: '#DDDDDD',
};

/**
 * Transform pixel to rem values
 * @param {number} px The pixel value to be transformed
 */
export const px2rem = (px) => `${px / 16}rem`;
