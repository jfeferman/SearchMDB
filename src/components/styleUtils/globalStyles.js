import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
   body {
    margin: 0;
    -webkit-font-smoothing: antialiased;
    -webkit-overflow-scrolling: touch;
  }

  html, body, input, select, option {
    font-family: 'Arial', sans-serif;
  }
`;
