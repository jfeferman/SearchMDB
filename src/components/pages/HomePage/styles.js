import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledResultsBlock = styled.div`
  margin: 0 auto;
  max-width: ${px2rem(980)};
  padding: 0 ${px2rem(40)} ${px2rem(80)} ${px2rem(40)};

  h3 {
    margin-bottom: ${px2rem(4)};
  }
`;
