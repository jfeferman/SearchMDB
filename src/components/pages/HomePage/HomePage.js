import React, { useState } from 'react';
import {
  SearchBlock,
  MovieCard,
  TvCard,
  PersonCard,
} from '../../organisms';
import { Heading } from '../../atoms';
import { GlobalStyle } from '../../styleUtils/globalStyles';
import { QueryContext } from './context';
import { multiQuery } from '../../../utils/apiQuery';
import { StyledResultsBlock } from './styles';
import { CONTENT_KEYS } from '../../../config/constants';

const HomePage = () => {
  // eslint-disable-next-line no-unused-vars
  const [data, setData] = useState({
    movies: [],
    tv: [],
    person: [],
  });

  async function handleQuery(query) {
    const res = await multiQuery(query);
    setData(res);
  }

  const movieCards = data.movies.map((item) => <MovieCard key={item.id} item={item} />);
  const tvCards = data.tv.map((item) => <TvCard key={item.id} item={item} />);
  const personCards = data.person.map((item) => <PersonCard key={item.id} item={item} />);

  return (
    <QueryContext.Provider
      value={{ onQuery: handleQuery }}
    >
      <GlobalStyle />
      <SearchBlock />
      <StyledResultsBlock>
        {data.movies.length > 0 && (
          <>
            <Heading level="2">{CONTENT_KEYS.HOME_MOVIES_HEADING}</Heading>
            {movieCards}
          </>
        )}
        {data.tv.length > 0 && (
          <>
            <Heading level="2">{CONTENT_KEYS.HOME_TV_HEADING}</Heading>
            {tvCards}
          </>
        )}
        {data.person.length > 0 && (
          <>
            <Heading level="2">{CONTENT_KEYS.HOME_PEOPLE_HEADING}</Heading>
            {personCards}
          </>
        )}
      </StyledResultsBlock>
    </QueryContext.Provider>
  );
};

export default HomePage;
