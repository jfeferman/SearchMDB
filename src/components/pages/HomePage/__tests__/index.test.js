import React from 'react';
import 'jest-styled-components';
import { create } from 'react-test-renderer';
import HomePage from '..';
import { SearchBlock } from '../../../organisms';
import { GlobalStyle } from '../../../styleUtils/globalStyles';

describe('components/organisms/SearchBlock', () => {
  let renderer;
  let instance;

  beforeEach(() => {
    renderer = create(<HomePage />);
    instance = renderer.root;
  });

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('contains the global styles', () => {
    expect(instance.findAll(el => el.type === GlobalStyle)).toHaveLength(1);
  });

  it('renders a search block', () => {
    expect(instance.findAll(el => el.type === SearchBlock)).toHaveLength(1);
  });
});
