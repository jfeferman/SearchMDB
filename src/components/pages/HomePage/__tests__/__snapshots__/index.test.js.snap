// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/SearchBlock matches the snapshot of the component 1`] = `
Array [
  .c4 {
  border-radius: 0.625rem;
  border: 1px solid;
  cursor: pointer;
  display: block;
  font-weight: 300;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  width: auto;
  background: #DDDDDD;
  color: #000000;
  font-size: 0.9375rem;
  line-height: 1.0625rem;
  -webkit-letter-spacing: 0.02125rem;
  -moz-letter-spacing: 0.02125rem;
  -ms-letter-spacing: 0.02125rem;
  letter-spacing: 0.02125rem;
  padding: 1rem 2.59375rem;
  vertical-align: middle;
  outline: 0;
  -webkit-transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out;
  transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out;
}

.c4:focus {
  box-shadow: 0px 0px 2px 2px rgba(23,109,220,0.8);
}

.c1 {
  font-weight: 300;
  font-style: normal;
  color: #000000;
  font-size: 2.8125rem;
  line-height: 3.875rem;
  -webkit-letter-spacing: 0.076875rem;
  -moz-letter-spacing: 0.076875rem;
  -ms-letter-spacing: 0.076875rem;
  letter-spacing: 0.076875rem;
}

.c3 {
  width: 100%;
  box-sizing: border-box;
  height: 3.125rem;
  border: 1px solid #DDDDDD;
  border-radius: 0.1875rem;
  background-color: #ffffff;
  padding-left: 1.25rem;
  font-size: 1rem;
  min-width: 25rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0 auto;
  width: 37.5rem;
}

.c2 div:nth-of-type(1) {
  margin-right: 0.625rem;
}

.c0 {
  margin: 6.25rem auto 6.25rem auto;
}

.c0 h1 {
  text-align: center;
}

@media (max-width:768px) {
  .c1 {
    font-size: 2.125rem;
    line-height: 3rem;
    -webkit-letter-spacing: 0.058125rem;
    -moz-letter-spacing: 0.058125rem;
    -ms-letter-spacing: 0.058125rem;
    letter-spacing: 0.058125rem;
  }
}

<div
    className="c0"
  >
    <h1
      className="c1"
    >
      Search MDB
    </h1>
    <div
      className="c2"
    >
      <div>
        <input
          className="c3"
          placeholder="Search for movie, tv show, person..."
          type="text"
        />
      </div>
      <div>
        <button
          className="c4"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>,
  .c0 {
  margin: 0 auto;
  max-width: 61.25rem;
  padding: 0 2.5rem 5rem 2.5rem;
}

.c0 h3 {
  margin-bottom: 0.25rem;
}

<div
    className="c0"
  />,
]
`;
